rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+
+  1. The hardening components require the use of a CIS-supplied base image that
+     incurrs an added per-minute royalty to CIS. As we are using this image as a
+     delivery channel to our customers, this is a non-starter.
+  2. The use of the CIS hardening tools requires a subscription to/partnreship with 
+     CIS that we do not currently have
+
+The decision to go with STIG is based on being able to improve the situation
+for our customers *now*, while still being able to add CIS hardening tools as
+we develop that relationship in the future.
+
+#### **Hardened Base Image**
+
+My proposed solution is to use AWS EC2 Image builder, using the available STIG
+tooling, to create a hardened base image. 
+
+We can do this as part of the build process, or on some frequent schedule 
+(e.g. dailiy, weekly).
+
+I suggest the latter, scheduled option;
+
+ 1. The release build process is already complicated enough, and
+ 2. There is no real advantage to creating a per-release base image, as there 
+    is no way to inject parameters into the image build pipeline.
+
+#### **Why ImageBuilder *AND* Packer?**
+
+As mentioned above, there is no way to inject parameters into an EC2 ImageBuilder
+pipeline. The pipeline is the pipeline. There is some customisation available at
+the build component level, but there is no way (that I can see, at least) to
+say something like "Trigger this pipeline using Teleport `vX.Y.Z` and save the
+resulting image as `teleport-oss-X.Y.Z`". All you can do is trigger the pipeline
+and have it produce (or replace) a single image.
+
+So this brings us to an obvious division of labour:
+
+ * **EC2 ImageBuilder** to build the hardened base image, which will be used as
+   the common base for all AMIs we produce, across all supported Teleport
+   versions, and
+ * **Packer** for the final, individual AMIs that are tailored to a specific
+   Teleport version and configuration.
+
+There is a side benefit here: were we to move the entire process into EC2 Image
+Builder, it would require moving a lot of Teleport-specific code and configuration
+*out* of the `teleport` Git repositories and into `teleport-cloud` terraform.
+
+From experience, keeping Teleport-specific things closer to `teleport` is generally
+a better idea than splitting them apart.
+
+#### **[Optional] Slimming the AMIs**
+
+In the spirit of moving towards distroless OCI container images, I propose slimming
+the contents of the AMIs in order to reduce their attack surface. 
+
+#### **Publishing to `teleport-prod`**
+
+These new images will be published to the `teleport-prod` AWS account. This will
+partiall obviate the need to move the legacy images from the `gravitational`
+AWS account. 
+
+As per the existing AMI builds, AMIs wull be constructed as part of the tag build, and
+made public on promotion.
+
+#### **Post-build validation using Amazon Inspector**
+
+While the CIS-hardening tooling on Amazon requires royalties and subscriptions,
```

Comment:
Did you mean to say STIG-hardening tooling here?

Reply:
I was trying to contrast the hardening tools with the validation tools. Will re-word for clarity